src/Microsoft.DotNet.Helix/Sdk/FindDotNetCliPackage.cs

```
@@ -102,31 +127,41 @@ private async Task ExecuteAsync()
+        private async Threading.Task<string> GetDownloadUrlAsync(ITaskItem feed)
         {
             var oldVersion = Version; // ResolveVersionAsync will adjust the Version property, but we need it set back for other feeds to see the same initial Version
             try
             {
                 var version = await ResolveVersionAsync(feed);
                 string extension = Runtime.StartsWith("win") ? "zip" : "tar.gz";
+
                 string effectiveVersion = await GetEffectiveVersion(feed, version);
+                feed.TryGetMetadata("SasToken", out string sasToken);
+                if (!string.IsNullOrEmpty(sasToken))
+                {
+                    if (!sasToken.StartsWith("?"))
+                    {
+                        sasToken = "?" + sasToken;
+                    }
+                }
 
                 return PackageType switch
                 {
-                    "sdk" => $"{feed}/Sdk/{version}/dotnet-sdk-{effectiveVersion}-{Runtime}.{extension}",
+                    "sdk" => $"{feed.ItemSpec}/Sdk/{version}/dotnet-sdk-{effectiveVersion}-{Runtime}.{extension}{sasToken}",
                     "aspnetcore-runtime" =>
-                        $"{feed}/aspnetcore/Runtime/{version}/aspnetcore-runtime-{effectiveVersion}-{Runtime}.{extension}",
-                    _ => $"{feed}/Runtime/{version}/dotnet-runtime-{effectiveVersion}-{Runtime}.{extension}"
+                        $"{feed.ItemSpec}/aspnetcore/Runtime/{version}/aspnetcore-runtime-{effectiveVersion}-{Runtime}.{extension}{sasToken}",
+                    _ => $"{feed.ItemSpec}/Runtime/{version}/dotnet-runtime-{effectiveVersion}-{Runtime}.{extension}{sasToken}"
                 };
             }
             catch (Exception ex)
             {
-                Log.LogWarning($"Unable to resolve download link from feed {feed}; {ex.Message}");
+                Log.LogWarning($"Unable to resolve download link from feed {feed}; {SanitizeString(ex.Message)}");
```

Why are we logging this as a warning and as a message?

---

i was hoping nobody would notice... is that a good reason?  I needed to put this warning in the messages so that i could get enough info to debug my test failures that were being reported back from Helix.  Sadly, this function swallows all the exceptions and puts them in the warnings block and i wasn't getting that dump from Helix.  i can refactor the test